### PR TITLE
Refactor logic around providing drug frequencies label depending on the country - Final

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://app.clubhouse.io/simpledotorg/story/4813/refactor-logic-around-providing-drug-frequencies-label-depending-on-the-country

### CODE REVIEW

**Comment by janhavisinghh** [comment]
Added @vinaysshenoy as an add on reviewer as this PR resolves the issues he raised in my [last](https://github.com/simpledotorg/simple-android/pull/3011) PR.

File: app/src/main/java/org/simple/clinic/drugs/selection/custom/drugfrequency/country/EthiopiaDrugFrequencyProvider.kt [hunk 1]
```diff
@@ -9,11 +9,11 @@ import org.simple.clinic.drugs.search.DrugFrequency.TDS
...
-        DrugFrequencyChoiceItem(drugFrequency = null, labelResId = R.string.custom_drug_entry_sheet_frequency_none),
-        DrugFrequencyChoiceItem(drugFrequency = OD, labelResId = R.string.custom_drug_entry_sheet_frequency_ethiopia_PD),
-        DrugFrequencyChoiceItem(drugFrequency = BD, labelResId = R.string.custom_drug_entry_sheet_frequency_ethiopia_BID),
-        DrugFrequencyChoiceItem(drugFrequency = TDS, labelResId = R.string.custom_drug_entry_sheet_frequency_ethiopia_TID),
-        DrugFrequencyChoiceItem(drugFrequency = QDS, labelResId = R.string.custom_drug_entry_sheet_frequency_ethiopia_QID)
+        DrugFrequencyChoiceItem(drugFrequency = null, label = "None"),
+        DrugFrequencyChoiceItem(drugFrequency = OD, label = "PD"),
+        DrugFrequencyChoiceItem(drugFrequency = BD, label = "BID"),
+        DrugFrequencyChoiceItem(drugFrequency = TDS, label = "TID"),
+        DrugFrequencyChoiceItem(drugFrequency = QDS, label = "QID")
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/drugs/selection/custom/drugfrequency/country/EthiopiaDrugFrequencyProvider.kt:16]: Why do we need to hard code the values in code? We can inject the `Resources` as a constructor parameter and use that to resolve the resources.

File: app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt [hunk 2]
```diff
@@ -29,11 +29,19 @@ class CustomDrugEntryUpdate : Update<CustomDrugEntryModel, CustomDrugEntryEvent,
         dispatch(RemoveDrugFromPrescription(update.prescribedDrugUuid))
       }
       is DrugFetched -> drugFetched(model, event.drug)
-      is DrugFrequencyChoiceItemsLoaded -> next(model.drugFrequencyChoiceItemsLoaded(event.drugFrequencyChoiceItems.items))
+      is DrugFrequencyChoiceItemsLoaded -> drugFrequencyChoiceItemsLoaded(model, event)
```
Comment (Honey14) [inline app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntryUpdate.kt:32]: We follow, `when..., then...` when we write commit messages. FYI, for future commits. We can merge this in. @janhavisinghh

**Review by Honey14 — APPROVED** [review]
LGTM

**Pull request merged**